opnode/node/driver.go

```diff
@@ -0,0 +1,87 @@
+package node
+
+import (
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/trie"
+	"github.com/holiman/uint256"
+)
+
+type InputItem struct {
+	Block    *types.Block
+	Receipts []*types.Receipt
+}
+
+func (item *InputItem) Epoch() uint64 {
+	return item.Block.NumberU64()
+}
+
+// sanity check that the receipts are consistent with the block data
+func (item *InputItem) CheckReceipts() bool {
+	hasher := trie.NewStackTrie(nil)
+	computed := types.DeriveSha(types.Receipts(item.Receipts), hasher)
+	return item.Block.ReceiptHash() == computed
+}
+
+type DepositTransaction struct {
+	Sender common.Address
+	// TODO some flag/bool for contract-deploy transaction. Or it can be *To with nil
+	Flags uint8
+	To    common.Address
+	Value uint256.Int
+	Data  []byte
+}
+
+func (dep *DepositTransaction) Decode(data []byte) (valid bool) {
+	// TODO
+	return false
+}
+
+type OpaqueTransaction []byte
+
+type OutputItem struct {
+	// TODO: convert to L1 attributes deposit
```

Comment:
Since we made the L1 attributes transaction a deposit-type transaction, shouldn't we bundle this along with `UserDepositTxs` (probably under a different name)?

Reply:
I kept them separate so we wouldn't have to define how it translates from typed properties to opaque encoded deposit transaction. In a follow-up PR we can implement actual block-derivation. And maybe we can do it in steps: decode into typed data first, then encode into a list of opaque transactions, and then complete the full block contents with execution through engine-api.